base/util.go

```diff
@@ -735,6 +735,24 @@ func CouchbaseURIToHttpURL(bucket Bucket, couchbaseUri string) (httpUrls []strin
 
 }
 
+// Add auth credentials to the given urls, since CBGT cannot take auth handlers in certain API calls yet
+func ServerUrlsWithAuth(urls []string, spec BucketSpec) (urlsWithAuth []string, err error) {
```

Comment:
Advisory:

Pre-allocate urlsWithAuth slice with length of input slice.

```suggestion
func ServerUrlsWithAuth(urls []string, spec BucketSpec) (urlsWithAuth []string, err error) {
	urlsWithAuth = make[]string, 0, len(urls))
```

Reply:
Avoiding repeated doubling of the slice capacity during append makes sense, and I agree we should do that generally.  In this particular case it's probably applicable to switch to using index positions instead of append altogether.